Gemfile

```diff
@@ -1,7 +1,7 @@
 source 'https://rubygems.org'
 
 gem 'appraisal', '2.2.0'
-gem 'bundler', '~> 1.1'
+gem 'bundler'
```

Comment:
I wonder if it still useful to keep Bundler 1.x for development. Every so often, I find that tests fail on Rails 4.2, so in order to debug this I will re-run the tests locally using the `rails_4_2` appraisal. I feel like Bundler 2.x doesn't work with Rails 4.2 (or it may be Ruby 2.4/2.5, as you put in your comment above, I'm not sure). So I feel like we might need to keep this? What do you think?

Reply:
Hi @mcmire 
Yeah, bundler 1.x is still needed for Rails 2.4 specs.
I removed the version restriction so the combination of Rails + Ruby can choose the most appropriate version.
Ruby 2.7 has no support for Bundler 1.x and Rails 4.2 requires `bundler < 2` in its `gemspec`:

https://github.com/rails/rails/blob/c0cb0cbf976a3cf8ad1b0e2d0f813602a712e997/rails.gemspec#L30

The solution was to only use Bundler 1.x for Ruby 2.4 and 2.5, supported by Rails 4.2:

https://github.com/thoughtbot/shoulda-matchers/blob/6757e3215a3dc17b0b5fe6653265f68554fb8cf8/.travis.yml#L32-L42

After these CI improvements, I feel like removing support for Ruby 2.4 and Rails 4.2. This would the codebase simpler and easier to contribute to. Some shims and ActiveRecord matches, especially `belongs_to` would benefit from that. But that's for a major release 😄

Reply:
Oh ok. So maybe this one's okay. What about the one in `rails_4_2.gemfile`, would you have to change that one back?